featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -651,6 +654,28 @@ def _feature_in_relationship_path(self, relationship_path, feature):
 
         return False
 
+    def _get_matching_inputs(self, all_features, entity, max_depth, input_types,
```

Comment:
creating aggregation features has a similar `_features_by_type` followed by `match` structure. should we try to get it to call this method too?